sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2006,15 +2006,19 @@ class Dataset[T] private[sql](
       }
       cols
     }
-    val groupColExprIds = groupCols.map(_.exprId)
-    val aggCols = logicalPlan.output.map { attr =>
-      if (groupColExprIds.contains(attr.exprId)) {
-        attr
-      } else {
-        Alias(new First(attr).toAggregateExpression(), attr.name)()
+    if (isStreaming) {
+      Deduplication(groupCols, logicalPlan)
+    } else {
+      val groupColExprIds = groupCols.map(_.exprId)
+      val aggCols = logicalPlan.output.map { attr =>
+        if (groupColExprIds.contains(attr.exprId)) {
+          attr
+        } else {
+          Alias(new First(attr).toAggregateExpression(), attr.name)()
```

Comment:
@marmbrus I tried to move this to SparkPlanner but failed because `Alias(new First(attr).toAggregateExpression(), attr.name)()` needs to be resolved before planning. Thoughts?